### PR TITLE
v1 of the code that enables the new editor for all users.

### PR DESCRIPTION
**Pull request by diegoreymendez**
Fixes [this issue](https://github.com/wordpress-mobile/WordPress-iOS-Editor/issues/497).
Fixes [this issue](https://github.com/wordpress-mobile/WordPress-iOS-Editor/issues/509).
Improves the code we had for [this issue](https://github.com/wordpress-mobile/WordPress-iOS-Editor/issues/466).

This was implemented using the minimum possible logic to turn the new editor on by default.  In my opinion we should consider in the next few versions cleaning up some of the code and removing the old visual editor - also removing some of the code that was introduced in this PR, since it won't be needed anymore.

How it works, please review the logic:
- **For users who didn't have the editor in earlier versions:** the new editor is enabled as default.  The onboarding tip is shown.
- **For users who had the editor in earlier versions:** the editor is still available in settings but not force enabled.  The onboarding tip is _not_ shown.

/cc @bummytime

### CODE REVIEW

File: WordPress/Classes/System/WordPressAppDelegate.m [hunk 1]
```diff
@@ -72,6 +72,7 @@
 
 int ddLogLevel                                                  = LOG_LEVEL_INFO;
 static NSString * const kUsageTrackingDefaultsKey               = @"usage_tracking_enabled";
+static NSString * const MustShowWhatsNewPopup = @"MustShowWhatsNewPopup";
```
Comment (bummytime) [inline WordPress/Classes/System/WordPressAppDelegate.m]: The fact this doesn't line up with the other const makes me cry. :crying_cat_face:  (kidding)

File: WordPress/Classes/ViewRelated/NUX/CreateAccountAndBlogViewController.m [hunk 2]
```diff
@@ -699,8 +699,6 @@ - (void)createUserAndSite
     WPAsyncBlockOperation *userCreation = [WPAsyncBlockOperation operationWithBlock:^(WPAsyncBlockOperation *operation){
         void (^createUserSuccess)(id) = ^(id responseObject){
             // Turn on the new editor only for users that create a new account within the iOS app
-            [WPPostViewController setNewEditorAvailable:YES];
-            [WPPostViewController setNewEditorEnabled:YES];
             [WPAnalytics track:WPAnalyticsStatEditorEnabledNewVersion];
```
Comment (bummytime) [inline WordPress/Classes/ViewRelated/NUX/CreateAccountAndBlogViewController.m]: Do we still need to log that the new editor enabled in analytics if it's on by default?

/cc @meremagee @sendhil

Comment (meremagee) [inline WordPress/Classes/ViewRelated/NUX/CreateAccountAndBlogViewController.m]: > Do we still need to log that the new editor enabled in analytics if it's on by default?

Yup - the goal is to get a feel for people who _toggle_ the editor on/off to compare high level perf stats for legacy vs new

Comment (bummytime) [inline WordPress/Classes/ViewRelated/NUX/CreateAccountAndBlogViewController.m]: @diegoreymendez @sendhil given what @meremagee said above, should we just add  ```[WPAnalytics track:WPAnalyticsStatEditorEnabledNewVersion];``` to ```setNewEditorEnabled``` in ```WPPostViewController```? And...do we need a new item in the ```WPAnalyticsStat``` enum to capture when the visual editor is turned off?

**Comment by bummytime** [comment]
@diegoreymendez The code looks good. Just had the analytics question above. Lemme know your thoughts.

File: WordPress/Classes/System/WordPressAppDelegate.m (hunk 1 above)
Comment (diegoreymendez) [inline WordPress/Classes/System/WordPressAppDelegate.m]: Fair enough, does look a bit off!

Changed it "Pfhor Your Eyes Only".  (If you happen to know that reference you have my respect O.O)

File: WordPress/Classes/ViewRelated/NUX/CreateAccountAndBlogViewController.m (hunk 2 above)
Comment (diegoreymendez) [inline WordPress/Classes/ViewRelated/NUX/CreateAccountAndBlogViewController.m]: @bummytime - I fixed the tracking as you proposed.  Good call!  Please let me know if there's anything else off.

**Comment by diegoreymendez** [comment]
@bummytime - Ready for round 2.

**Comment by bummytime** [comment]
Love it! :shipit:

**Pull request merged**